source/ed/ed_reader.cpp

```diff
@@ -775,8 +774,20 @@ void EdReader::fill_vehicle_journeys(nt::Data& data, pqxx::work& work){
         vj->vehicle_journey_type = static_cast<nt::VehicleJourneyType>(const_it["odt_type_id"].as<int>());
         vj->journey_pattern = journey_pattern;
 
-        vj->company = company_map[const_it["company_id"].as<idx_t>()];
 
+        vj->company = company_map[const_it["company_id"].as<idx_t>()];
+        if (vj->company && vj->journey_pattern && vj->journey_pattern->route && vj->journey_pattern->route->line){
+            if(std::find(vj->journey_pattern->route->line->company_list.begin(),
```

Comment:
prefer the usage of [boost::find](http://www.boost.org/doc/libs/1_46_1/libs/range/doc/html/range/reference/algorithms/non_mutating/find.html)